### PR TITLE
fix: expand extract_error_message Jira error format coverage

### LINKED ISSUE #171: extract_error_message: expand Jira error format coverage
## Problem

`extract_error_message` at `src/api/client.rs` only handles two Jira error response formats:

1. `errorMessages` array (non-empty) → joined with "; "
2. `message` string field

Real-world Jira responses use several other formats that currently fall through to raw body dump or produce a useless error:

1. **`errors` object (field-level validation)** — `{"errorMessages":[],"errors":{"summary":"is required"}}` — the most common JSM/Jira validation error response. Falls through to raw body.
2. **`errorMessage` singular** (JSM-specific) — `{"errorMessage":"Cannot find issue"}`. Falls through to raw body.
3. **`status-code` + `message`** — `{"status-code":404,"message":"Property with key not found"}`. The `message` extraction works but the format is documented separately.
4. **Empty body** — 5xx responses with no body produce `Error: API error (500):` (bare trailing colon) via \`JrError::ApiError\` Display.

## Impact

When using \`jr api -X POST /rest/api/3/issue -d '{...}'\` with a missing field, the user sees the entire raw JSON dumped on stderr instead of a clean "summary: is required" message.

Identified during PR #TBD review of #111 (raw API passthrough). Pre-existing gap in \`parse_error\`, now more user-visible via \`jr api\` since the helper is \`pub\`.

## Proposed

Expand \`extract_error_message\` to handle:
- Non-empty \`errors\` object → format as \`field1: msg1; field2: msg2\`
- \`errorMessage\` singular → return the string
- Empty body → return a sentinel like \`"<empty response body>"\`
- Non-UTF-8 body → prefix with \`"[non-UTF-8 response body]: ..."\`

Add unit tests for each.

## Reference

- Documented as a caveat in \`docs/superpowers/specs/2026-04-08-api-passthrough-design.md\` (Caveats section)
- Discovered via pr-review-toolkit analysis of PR implementing #111

### PR DESCRIPTION
**Pull request by Zious11**
## Summary

Closes #171. Expands `extract_error_message` in `src/api/client.rs` to handle additional Jira Cloud error response formats that previously fell through to raw body dump.

- **`errors` object** (field-level validation): `{"errorMessages":[],"errors":{"summary":"is required"}}` → `"summary: is required"`. Non-string values (nested objects, arrays) are stringified rather than dropped.
- **`errorMessage` singular** (defensive for JSM endpoints): `{"errorMessage":"Cannot find issue"}` → `"Cannot find issue"`
- **Empty body** → `"<empty response body>"` sentinel instead of bare empty string
- Precedence chain: `errorMessages` array → `errors` object → `message` → `errorMessage` → empty sentinel → raw body fallback
- Updated api-passthrough spec caveat to reflect expanded coverage

## Test Plan

- [x] `cargo test` — 578 tests pass (13 `extract_error_message` tests: 6 existing + 5 new formats + 2 non-string value edge cases)
- [x] `cargo clippy -- -D warnings` — clean
- [x] `cargo fmt --all -- --check` — clean
- [x] PR review (code-reviewer + silent-failure-hunter) — addressed Important finding (non-string errors values)

### CODE REVIEW

File: tests/api_client.rs [hunk 1]
```diff
@@ -119,10 +119,62 @@ fn test_extract_error_message_prefers_error_messages_over_message() {
...
+    let result = extract_error_message(body);
+    // Both fields present, joined with "; " (order may vary due to JSON object)
+    assert!(
+        result.contains("summary: is required"),
+        "expected 'summary: is required' in '{result}'"
+    );
+    assert!(
+        result.contains("priority: is required"),
+        "expected 'priority: is required' in '{result}'"
+    );
```
[the remark this answers is not in the document]
Comment (Zious11) [reply tests/api_client.rs]: Fixed in 4698095. Replaced `contains()` assertions with exact `assert_eq!` since `pairs.sort()` guarantees deterministic output. Removed the misleading comment about order variance.

File: docs/superpowers/specs/2026-04-08-api-passthrough-design.md [hunk 2]
```diff
@@ -308,7 +308,7 @@ All JSON is synthetic. No real project keys, org IDs, account IDs, or instance U
 - **Body size:** The entire body is read into memory before sending. Not suitable for very large payloads (multi-MB uploads) — but Jira's standard API is not typically used for large payloads.
 - **Streaming responses:** `jr api` reads the entire response into memory before printing. Fine for JSON payloads; not suitable for streaming endpoints (Jira has none in practice).
 - **`@` prefix in filenames:** A filename literally starting with `@` must be passed as `./@file.json` to avoid being interpreted as a nested reference. Matches curl's behavior.
-- **Error message extraction is incomplete (pre-existing):** `extract_error_message` preserves the existing `parse_error` behavior — it handles `errorMessages` array and `message` string but not `errorMessage` (singular JSM field), `errors` object (field-level validation), or `status-code` + `message` format. Expanding format coverage is tracked as a follow-up enhancement, not in scope for this PR.
+- **Error message extraction:** `extract_error_message` now handles `errorMessages` array, `errors` object (field-level validation), `message` string, `errorMessage` singular, and empty bodies. See #171.
```
[the remark this answers is not in the document]
Comment (Zious11) [reply docs/superpowers/specs/2026-04-08-api-passthrough-design.md:311]: Fixed in 4698095. Updated the Client Tests table at line 292 to show `"" → "<empty response body>"` matching the new sentinel behavior.

File: docs/superpowers/specs/2026-04-08-api-passthrough-design.md [hunk 3]
```diff
@@ -289,7 +289,7 @@ assert_eq!(content_type_count, 1);
 | `test_extract_error_message_from_error_messages` | `{"errorMessages":["foo","bar"]}` → `"foo; bar"` |
 | `test_extract_error_message_from_message_field` | `{"message":"foo"}` → `"foo"` |
 | `test_extract_error_message_from_plain_text` | `"not json"` → `"not json"` (fallback) |
```
[the remark this answers is not in the document]
Comment (Zious11) [reply docs/superpowers/specs/2026-04-08-api-passthrough-design.md]: Fixed in 0aa7e14. Updated all three mismatched test names in the Client Tests table to match the actual function names in tests/api_client.rs.

File: docs/superpowers/specs/2026-04-08-api-passthrough-design.md [hunk 4]
```diff
@@ -286,10 +286,10 @@ assert_eq!(content_type_count, 1);
 | `test_send_raw_returns_response_for_404` | `send_raw` returns `Response` (NOT an error) for 404 — critical for raw passthrough |
 | `test_send_raw_retries_429` | `send_raw` retries 429 with `Retry-After`, then returns 200 response |
 | `test_send_raw_returns_response_after_exhausted_429` | After `MAX_RETRIES` 429s, returns the 429 `Response` (caller decides what to do) |
```
[the remark this answers is not in the document]
Comment (Zious11) [reply docs/superpowers/specs/2026-04-08-api-passthrough-design.md]: Fixed in de4ff9e. Updated both send_raw 429 test names in the Client Tests table to match the actual function names.
